record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/CollectionBuilderUtils.java

```diff
@@ -298,7 +300,13 @@ private String disambiguateGeneratedMethodName(List<RecordClassType> recordCompo
 
     private MethodSpec buildShimMethod(String name, TypeName mainType, Class<?> abstractType,
             ParameterizedTypeName parameterizedType, TypeVariableName... typeVariables) {
-        var code = CodeBlock.of("return (o != null) ? $T.copyOf(o) : $T.of()", mainType, mainType);
+        CodeBlock code;
+        if (mainType.equals(setTypeName)) {
```

Comment:
Are there backward compatibility concerns with this? We should probably put this behind an option. wdyt?